### PR TITLE
Issue #2483 - Add backendtlspolicy API Documentation

### LINKED ISSUE #2483: Documentation for BackendTLSPolicy
**What would you like to be added**:

- [x] Update https://gateway-api.sigs.k8s.io/guides/tls/ with BackendTLSPolicy

If time permits, 
- [x] Add API type BackendTLSPolicy

**Why this is needed**:
Needed to document new feature in release 1.0

### PR DESCRIPTION
**Pull request by candita**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
Add further documentation for new feature BackendTLSPolicy.

**Which issue(s) this PR fixes**:

Fixes #2483

**Does this PR introduce a user-facing change?**:

```release-note
Document BackendTLSPolicy.
```

### CODE REVIEW

**Review by kflynn — APPROVED** [review]
Thanks @candita! These look good from here.

File: site-src/api-types/backendtlspolicy.md
Comment (mikemorris) [inline site-src/api-types/backendtlspolicy.md:33]: Does this imply "all Routes [attached to a Gateway with support for BackendTLSPolicy] that point to a referenced Service should/must respect a configured policy"?

I think this could be slightly more clear on whether respecting this configuration is optional ("one can reuse") or expected in scenarios with multiple Routes.

**Review by mikemorris — APPROVED** [review]
LGTM, excited to see this land!

As a non-blocking suggestion, we may want to add a section to https://gateway-api.sigs.k8s.io/guides/tls/ directing users to this documentation, and update the terminology on that page to be consistent with the terms used here (I think "edge" and "backend" are more clear than that page's use of "downstream" and "upstream").

**Review by shaneutt — APPROVED** [review]
/approve

Looks good!

@robscott should check this out before we merge.

/hold

**Comment by robscott** [comment]
Trying to kick netlify to get deploy previews going, will reopen shortly.

/close

File: site-src/api-types/backendtlspolicy.md
Comment (robscott) [reply site-src/api-types/backendtlspolicy.md:33]: Good suggestion @mikemorris, maybe something like this? Once this graduates to standard channel, I think "Should" will become "Must" here.
```suggestion
All Gateway API Routes that point to a referenced Service should respect a configured BackendTLSPolicy.
```

File: site-src/api-types/backendtlspolicy.md
Comment (robscott) [inline site-src/api-types/backendtlspolicy.md:74]: Would recommend without backticks, but that's non-blocking. In either case, looks like another space was needed here.
```suggestion
A BackendTLSPolicy targets a backend Pod (or set of Pods) via a TargetRef to a Service.  This TargetRef is a
```

File: site-src/api-types/backendtlspolicy.md
Comment (robscott) [inline site-src/api-types/backendtlspolicy.md:79]: Hard to tell without working deploy preview, but I think you may need a new line + indentation here to match what you have below. 
```suggestion

    - Cross-namespace certificate references are not allowed.
```

File: site-src/api-types/backendtlspolicy.md
Comment (robscott) [inline site-src/api-types/backendtlspolicy.md:48]: This diagram is a bit confusing to me. Maybe it's just that it's hard to distinguish the field name with the field type. I'd consider omitting it for now and adding it in a follow up. For any diagrams that aren't just mermaid.js, is it possible to share a link to a source file that others can copy to modify?

**Review by robscott — COMMENTED** [review]
Thanks @candita! A few nits on this one, but otherwise LGTM. Hopefully the deploy preview starts working on this PR sometime soon, but if not, we can just have a follow up to catch any issues.

File: site-src/api-types/backendtlspolicy.md
Comment (candita) [reply site-src/api-types/backendtlspolicy.md]: Agreed. Thanks!

File: site-src/api-types/backendtlspolicy.md
Comment (candita) [reply site-src/api-types/backendtlspolicy.md:43]: I'm going to share some modified mermaid for this diagram.  I can still omit it if you think it doesn't add any value.  There is one other diagram here (TLS Termination Types) that exists already in the repo.  I can try to find the original source.

**Comment by candita** [comment]
> As a non-blocking suggestion, we may want to add a section to https://gateway-api.sigs.k8s.io/guides/tls/ directing users to this documentation, and update the terminology on that page to be consistent with the terms used here (I think "edge" and "backend" are more clear than that page's use of "downstream" and "upstream").  

Good idea. @robscott suggested something similar.  I'll follow up when both of these are merged into main.  https://github.com/kubernetes/website/issues/43776

File: site-src/api-types/backendtlspolicy.md
Comment (candita) [reply site-src/api-types/backendtlspolicy.md]: @robscott I also just updated the mkdocs.yaml and alphabetized the API types there.  I didn't want BackendTLSPolicy to be the first in the list, so I added a suboption called Policy, where we can put more Policies in the future.

**Comment by robscott** [comment]
Thanks @candita!

/lgtm
/hold cancel

File: site-src/api-types/backendtlspolicy.md
Comment (robscott) [inline site-src/api-types/backendtlspolicy.md:117]: Potential follow up - this is currently styled in all caps (https://deploy-preview-2547--kubernetes-sigs-gateway-api.netlify.app/api-types/backendtlspolicy/). It might be better to just go with bold text here instead of a heading to avoid that styling.

**Pull request merged**
